tests/integration_tests/ui_tests/ArelleGUITest/ArelleGUITest/Tests.cs

```diff
@@ -0,0 +1,373 @@
+            {
+                var sb = new StringBuilder($"name='{d.Name}' type={d.ControlType}");
+                if (d.ControlType == ControlType.Window)
+                {
+                    sb.Append($" modal={d.AsWindow().IsModal}");
+                }
+                Logger.Default.Debug(sb.ToString());
+            }
+        }
+
+        [Test]
+        public void TestOpen()
+        {
+            var title = Retry.WhileEmpty(() => Window.Title).Result;
+            Assert.That(Window.Title, Does.Contain("arelle"));
+        }
+
+        [Test]
+        public void TestLoadDocument()
+        {
+            Logger.Default.Info("start TestLoadDocument");
+            Keyboard.TypeSimultaneously(VirtualKeyShort.CONTROL, VirtualKeyShort.KEY_O);
+            Wait.UntilInputIsProcessed();
+            Logger.Default.Info("find the Open File dialog");
+            // more reliable than Window.ModalWindows because that searches all descendants whereas this only searches children
+            var openFileWindow = Retry.WhileNull(() => Window.FindFirstChild(cf =>
+                cf.ByControlType(ControlType.Window).And(new PropertyCondition(Automation.PropertyLibrary.Window.IsModal, true))),
+                timeout: TimeSpan.FromSeconds(5)).Result;
+            Assert.That(openFileWindow, Is.Not.Null);
+
+            var exampleDocumentPath = Path.Join(ExecutionType.ArelleResourcesPath, "workiva.zip");
+            var exampleDocumentEntryPoint = "wk-20220331.htm";
+            int nFilesInDocumentZip, indexOfEntryPoint;
+            using (var archive = ZipFile.OpenRead(exampleDocumentPath))
+            {
+                nFilesInDocumentZip = archive.Entries.Count;
+                indexOfEntryPoint = archive.Entries.Select((e, i) => new { e, i }).Where(x => x.e.Name == exampleDocumentEntryPoint).Single().i;
+            }
+            var fileNameComboBox = Retry.WhileNull(
+                () => openFileWindow.FindFirstDescendant(cf => cf.ByControlType(ControlType.ComboBox).And(cf.ByName("File name:")))).Result.AsComboBox();
+            Logger.Default.Info("enter path to document");
+            fileNameComboBox.EditableText = exampleDocumentPath;
+            Wait.UntilInputIsProcessed();
+            Logger.Default.Info("open document");
+            Keyboard.Type(VirtualKeyShort.ENTER);
+            Wait.UntilInputIsProcessed();
+
+            Logger.Default.Info("select file inside document archive");
+            var selectFileFromArchiveWindow = Retry.WhileNull(
+                () => Window.FindFirstChild(Automation.ConditionFactory.ByControlType(ControlType.Window)), timeout: TimeSpan.FromSeconds(3)).Result;
+            var selectFileFromArchivePane = selectFileFromArchiveWindow.FindFirstByXPath("/Pane/Pane/Pane/Pane[1]");
+            Mouse.LeftClick(new Point(
+                (int)(selectFileFromArchivePane.BoundingRectangle.X + selectFileFromArchivePane.BoundingRectangle.Width / 2.0f),

```

Comment:
Maybe abstract to a click in center that takes the thing that you want to click?